exporter/loggingexporter/logging_exporter.go

```diff
@@ -25,18 +28,128 @@ import (
+}
+
+func (b *traceDataBuffer) logAttr(label string, value string) {
+	b.logEntry("    %-12s: %s", label, value)
+}
+
+func (b *traceDataBuffer) logMap(label string, data *map[string]string) {
+	if data == nil {
+		return
+	}
+
+	b.logEntry("%s:", label)
+	for label, value := range *data {
+		b.logEntry("     -> %s: %s", label, value)
+	}
+}
+
+type loggingExporter struct {
+	level  string
+	logger *zap.Logger
+}
+
+func (s *loggingExporter) pushTraceData(
+	ctx context.Context,
+	td consumerdata.TraceData,
+) (int, error) {
+	debug := s.level == "debug"
+
+	buf := traceDataBuffer{}
+
+	buf.logEntry("TraceData with %d spans", len(td.Spans))
+
+	if td.Resource != nil {
+		buf.logEntry("Resource %s with %d labels", td.Resource.Type, len(td.Resource.Labels))
+		if debug {
+			buf.logMap("Resource labels", &td.Resource.Labels)
+		}
+	}
+
+	if debug {
+		if td.Node != nil {
+			buf.logEntry("Node service name: %s", td.Node.ServiceInfo.Name)
+			id := td.Node.Identifier
+			if id != nil {
+				buf.logEntry("HostName: %s", id.HostName)
+				buf.logEntry("PID %d", id.Pid)
+			}
+			buf.logMap("Node attributes", &td.Node.Attributes)
+			li := td.Node.LibraryInfo
+			if li != nil {
+				buf.logEntry("Library language: %s", li.Language.String())
+				buf.logEntry("Core library version: %s", li.CoreLibraryVersion)
+				buf.logEntry("Exporter version: %s", li.ExporterVersion)
+			}
+		}
+	}
+
+	s.logger.Info(buf.str.String())
+
+	if debug {
+		for i, span := range td.Spans {
+			buf = traceDataBuffer{}
+			buf.logEntry("Span #%d", i)
+			if span == nil {
+				buf.logEntry("* Empty span")
+				continue
+			}
+
+			buf.logAttr("Trace ID", hex.EncodeToString(span.TraceId))
+			buf.logAttr("ID", hex.EncodeToString(span.SpanId))
+			buf.logAttr("Parent ID", hex.EncodeToString(span.ParentSpanId))
+			buf.logAttr("Name", span.Name.Value)
+			buf.logAttr("Kind", span.Kind.String())
+			buf.logAttr("Start time", span.StartTime.String())
+			buf.logAttr("End time", span.EndTime.String())
+			if span.Status != nil {
+				buf.logAttr("Status code", string(span.Status.Code))
+				buf.logAttr("Status message", span.Status.Message)
+			}
+
+			if span.Attributes != nil {
+				buf.logAttr("Span attributes", "")
+				for attr, value := range span.Attributes.AttributeMap {
+					v := ""
+					ts := value.GetStringValue()
+
+					if ts != nil {
+						v = ts.Value
+					} else {
+						// For other types, just use the proto compact form rather than digging into series of checks
+						v = value.String()
+					}
+
+					buf.logEntry("         -> %s: %s", attr, v)
+				}
+			}
+
+			s.logger.Debug(buf.str.String())
+		}
+	}
+
+	return 0, nil
+}
+
 // NewTraceExporter creates an exporter.TraceExporter that just drops the
 // received data and logs debugging messages.
-func NewTraceExporter(config configmodels.Exporter, logger *zap.Logger) (exporter.TraceExporter, error) {
-	typeLog := zap.String("type", config.Type())
-	nameLog := zap.String("name", config.Name())
```

Comment:
Did we lose ability to see the type and name or this code is moved to somewhere?

Reply:
My take on that was that adding `type":"logging","name":"logging"` was not very helpful. I think its purpose was to aid with identifying the logs as logging exporter-ones, but since it's pretty obvious when looking at those (also because it's using different encoding now), it adds no value, so I removed it.

There's a good chance I might have missed something though, if that's the case then I will add them back

Reply:
I'd leave at least the name. If you have multiple pipelines with logging exporters you may want to know which one is output the logs.

Reply:
Suggestion: use `zap.String("exporter", config.Name()`